resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsWatchSnapshotSource.scala

```diff
@@ -78,6 +80,15 @@ class ExecutorPodsWatchSnapshotSource(
     }
   }
 
+  def reset(watcher: ExecutorPodsWatcher): Unit = {
```

Comment:
Have you ever test it?
The problem is that kubernetes client has been closed. reset may throw an exception.

Reply:
@sharkdtu What kind of exception would you expect? We're running this patch at the moment and can check.